lib/pos_mempool.go

```diff
@@ -277,14 +289,35 @@ func (mp *PosMempool) Start() error {
 			return errors.Wrapf(err, "PosMempool.Start: Problem loading persisted transactions")
 		}
 	}
+	mp.startGroup.Add(2)
+	mp.exitGroup.Add(2)
+	mp.startTransactionValidationRoutine()
 	mp.startAugmentedViewRefreshRoutine()
-
+	mp.startGroup.Wait()
 	mp.status = PosMempoolStatusRunning
 	return nil
 }
 
+func (mp *PosMempool) startTransactionValidationRoutine() {
+	go func() {
+		mp.startGroup.Done()
+		for {
+			select {
+			case <-time.After(time.Duration(mp.transactionValidationRefreshIntervalMillis) * time.Millisecond):
+				if err := mp.validateTransactions(); err != nil {
+					glog.Errorf("PosMempool.startTransactionValidationRoutine: Problem validating transactions: %v", err)
+				}
+			case <-mp.quit:
+				mp.exitGroup.Done()
+				return
+			}
+		}
+	}()
+}
+
 func (mp *PosMempool) startAugmentedViewRefreshRoutine() {
 	go func() {
+		mp.startGroup.Done()
```

Comment:
The logic in this function seems to implement almost 1:1 what you're doing in ConnectTransactions\[WithLimit\]. Since you have to do some flavor of `connect all these transactions to a view`, I would consider a change as follows:
Instead of having ConnectTransactions ON the view, move it to pos_mempool.go over here with a signature something like the following:
```
func ConnectTransactionsToViewWithLimit(oldView *UtxoView, txnsToConnect []*MempoolTxn, numTxnsLimit int, ignoreFailing bool, etc...)
```
Then have this function simply return a new view that has connected everything to a copy of oldView. And you can use that function here and in the place where you're currently using COnnectTransactions. The upshot of this approach is you avoid adding code to the view and, critically, you only do ONE connect per txn because you don't need to update the internal state of a view.